File: wiz-event-mall-java/src/main/java/com/example/wizeventmalljava/api/member/dto/MemberRegisterRequest.java
package com.example.wizeventmalljava.api.member.dto;

import com.example.wizeventmalljava.domain.member.entity.Member;

public record MemberRegisterRequest(
	String email,
	String password,
	String nickname
) {

	public Member toEntity() {
		return Member.builder()
			.email(email)
			.password(password)
			.nickname(nickname)
			.build();
	}
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/api/member/dto/MemberRegisterRequest.kt
package com.example.wizeventmall.api.member.dto

import com.example.wizeventmall.domain.member.entity.Member

data class MemberRegisterRequest(
    val email: String,
    val password: String,
    val nickname: String
) {

    fun toEntity() = Member(
        email = this.email,
        password = this.password,
        nickname = this.nickname
    )
}